features/base/test/test_metadata_connection.py

```diff
@@ -5,7 +5,9 @@
 def test_metadata_connection(client, non_azure, non_ali, non_chroot, non_kvm):
     metadata_host = "169.254.169.254"
     (exit_code, output, error) = client.execute_command(
-        f"wget --timeout 5 http://{metadata_host}"
+        f"wget --timeout 5 \
+               --header=\"X-aws-ec2-metadata-token: $(curl -sqX PUT 'http://{metadata_host}/latest/api/token' -H 'X-aws-ec2-metadata-token-ttl-seconds: 21600')\" \
```

Comment:
From the perspective of readability, i would prefer not to get this token value in a subshell command.
Any reason against splitting this into two? Might make it more readable and more easy to debug in case of failure. Also this can include an assertion on the expected return value of the `api/token` endpoint.